src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User register(String login, String password) throws RegistrationException {
+        Optional<User> userFromDB = userService.findByLogin(login);
+        if (userFromDB.isPresent()) {
+            throw new RegistrationException("Login " + login + " already in use");
+        }

```

Comment:
There is no need to create a named object if we use it only once
```suggestion
        if (userService.findByLogin(login).isPresent()) {
            throw new RegistrationException("Login " + login + " already in use");
        }

```